service/hook/gogs/gogs.go

```diff
@@ -86,13 +91,49 @@ func transformCodePushEvent(codePushEvent CodePushEventModel) hookCommon.Transfo
 		}
 	}
 
+	if len(lastCommit.CommitHash) == 0 {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Missing commit hash"),
+		}
+	}
+
+	if !strings.HasPrefix(pushEvent.Ref, "refs/heads/") {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Ref (%s) is not a head ref", pushEvent.Ref),
+		}
+	}
+
+	branch := strings.TrimPrefix(pushEvent.Ref, "refs/heads/")
+
 	return hookCommon.TransformResultModel{
 		TriggerAPIParams: []bitriseapi.TriggerAPIParamsModel{
 			{
 				BuildParams: bitriseapi.BuildParamsModel{
+					Branch:        branch,
 					CommitHash:    lastCommit.CommitHash,
 					CommitMessage: lastCommit.CommitMessage,
-					Branch:        branch,
+				},
+			},
+		},
+	}
+}
+
+func transformCreateEvent(createEvent CreateEventModel) hookCommon.TransformResultModel {
+	// Currently, we only support tag creation, not branches. Even if we wanted branch creation
+	// to trigger a build we would have to wait for https://github.com/go-gitea/gitea/issues/286
+	// to be in both Gogs and Gitea core, and well adopted/distributed.
+	if createEvent.RefType != "tag" {
+		return hookCommon.TransformResultModel{
+			Error:      errors.New("Ignoring branch-create request"),
```

Comment:
I'd change this message to "Not a tag create event - ignoring" or something like that; that describes more precisely what happens.